File: examples/sync/gateio_public.py
import dcex


def main() -> None:
    client = dcex.gateio()

    contracts = client.get_all_futures_contracts(ccy="usdt")
    print(contracts)

    orderbook = client.get_spot_order_book(product_symbol="BTC-USDT-SPOT")
    print(orderbook)


if __name__ == "__main__":
    main()

File: crates/dcex/examples/gateio_public.rs
use std::time::Duration;

use dcex::exchanges::gateio::GateioClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = GateioClient::public(Duration::from_secs(10))?;
    let response = client.get_spot_order_book("BTC-USDT-SPOT").limit(5).await?;
    println!("{}", response.data);
    Ok(())
}
